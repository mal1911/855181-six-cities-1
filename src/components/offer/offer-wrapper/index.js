import OfferWrapper from './offer-wrapper';

export default OfferWrapper;
