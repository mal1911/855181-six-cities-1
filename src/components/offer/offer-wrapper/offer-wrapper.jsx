import React from "react";
import {offerType} from "../../../prop-types";
import {getRandomArray} from "../../../util";
import OfferComments from "../offer-comments";

const OfferWrapper = ({offerObj}) => {
  const MAX_OFFER_IMAGES = 6;

  const imgs = getRandomArray(offerObj.images, MAX_OFFER_IMAGES).map((imageSrc, index) => <div key={index} className="property__image-wrapper">
    <img className="property__image" src={imageSrc} alt="Photo studio"/>
  </div>);

  const isPremium = () => {
    return offerObj.isPremium
      ? <div className="property__mark">
        <span>Premium</span>
      </div>
      :
      null;
  };
  const rating = Math.round(offerObj.rating);
  const insides = offerObj.goods.map((good, index) => <li key={index} className="property__inside-item">{good}</li>);
  const userStatus = offerObj.host.isPro ? `Pro` : ``;
  const userStatusClassName = offerObj.host.isPro ? `property__avatar-wrapper--pro` : ``;

  return <main className="page__main page__main--property">
    <section className="property">
      <div className="property__gallery-container container">
        <div className="property__gallery">
          {imgs}
        </div>
      </div>;
      <div className="property__container container">
        <div className="property__wrapper">
          {isPremium()}
          <div className="property__name-wrapper">
            <h1 className="property__name">
              {offerObj.title}
            </h1>
            <button className="property__bookmark-button button" type="button">
              <svg className="property__bookmark-icon" width="31" height="33">
                <use xlinkHref="#icon-bookmark"></use>
              </svg>
              <span className="visually-hidden">To bookmarks</span>
            </button>
          </div>
          <div className="property__rating rating">
            <div className="property__stars rating__stars">
              <span style={{width: `${rating * 20}%`}}></span>
              <span className="visually-hidden">Rating</span>
            </div>
            <span className="property__rating-value rating__value">{rating}</span>
          </div>
          <ul className="property__features">
            <li className="property__feature property__feature--entire">
              Entire {offerObj.type}
            </li>
            <li className="property__feature property__feature--bedrooms">
              {offerObj.bedrooms} Bedrooms
            </li>
            <li className="property__feature property__feature--adults">
              Max {offerObj.maxAdults} adults
            </li>
          </ul>
          <div className="property__price">
            <b className="property__price-value">&euro;{offerObj.price}</b>
            <span className="property__price-text">&nbsp;night</span>
          </div>
          <div className="property__inside">
            <h2 className="property__inside-title">What&apos;s inside</h2>
            <ul className="property__inside-list">{insides}</ul>
          </div>
          <div className="property__host">
            <h2 className="property__host-title">Meet the host</h2>
            <div className="property__host-user user">
              <div className={`property__avatar-wrapper ${userStatusClassName} user__avatar-wrapper`}>
                <img className="property__avatar user__avatar" src={`/${offerObj.host.avatarUrl}`} width="74" height="74" alt="Host avatar"/>
              </div>
              <span className="property__user-name">{offerObj.host.name}</span>
              <span className="property__user-status">{userStatus}</span>
            </div>
            <div className="property__description">
              <p className="property__text">{offerObj.description}</p>
            </div>
          </div>
          <OfferComments offerObj={offerObj} />
        </div>
      </div>
      <section className="property__map map"></section>
    </section>
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <div className="near-places__list places__list">
          <article className="near-places__card place-card">
            <div className="near-places__image-wrapper place-card__image-wrapper">
              <a href="#">
                <img className="place-card__image" src="img/room.jpg" width="260" height="200" alt="Place image"/>
              </a>
            </div>
            <div className="place-card__info">
              <div className="place-card__price-wrapper">
                <div className="place-card__price">
                  <b className="place-card__price-value">&euro;80</b>
                  <span className="place-card__price-text">&#47;&nbsp;night</span>
                </div>
                <button className="place-card__bookmark-button place-card__bookmark-button--active button" type="button">
                  <svg className="place-card__bookmark-icon" width="18" height="19">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">In bookmarks</span>
                </button>
              </div>
              <div className="place-card__rating rating">
                <div className="place-card__stars rating__stars">
                  <span style={{width: `80%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
              </div>
              <h2 className="place-card__name">
                <a href="#">Wood and stone place</a>
              </h2>
              <p className="place-card__type">Private room</p>
            </div>
          </article>

          <article className="near-places__card place-card">
            <div className="near-places__image-wrapper place-card__image-wrapper">
              <a href="#">
                <img className="place-card__image" src="img/apartment-02.jpg" width="260" height="200" alt="Place image"/>
              </a>
            </div>
            <div className="place-card__info">
              <div className="place-card__price-wrapper">
                <div className="place-card__price">
                  <b className="place-card__price-value">&euro;132</b>
                  <span className="place-card__price-text">&#47;&nbsp;night</span>
                </div>
                <button className="place-card__bookmark-button button" type="button">
                  <svg className="place-card__bookmark-icon" width="18" height="19">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="place-card__rating rating">
                <div className="place-card__stars rating__stars">
                  <span style={{width: `80%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
              </div>
              <h2 className="place-card__name">
                <a href="#">Canal View Prinsengracht</a>
              </h2>
              <p className="place-card__type">Apartment</p>
            </div>
          </article>

          <article className="near-places__card place-card">
            <div className="near-places__image-wrapper place-card__image-wrapper">
              <a href="#">
                <img className="place-card__image" src="img/apartment-03.jpg" width="260" height="200" alt="Place image"/>
              </a>
            </div>
            <div className="place-card__info">
              <div className="place-card__price-wrapper">
                <div className="place-card__price">
                  <b className="place-card__price-value">&euro;180</b>
                  <span className="place-card__price-text">&#47;&nbsp;night</span>
                </div>
                <button className="place-card__bookmark-button button" type="button">
                  <svg className="place-card__bookmark-icon" width="18" height="19">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="place-card__rating rating">
                <div className="place-card__stars rating__stars">
                  <span style={{width: `100%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
              </div>
              <h2 className="place-card__name">
                <a href="#">Nice, cozy, warm big bed apartment</a>
              </h2>
              <p className="place-card__type">Apartment</p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>;
};

OfferWrapper.propTypes = {
  offerObj: offerType.isRequired,
};

export default OfferWrapper;
