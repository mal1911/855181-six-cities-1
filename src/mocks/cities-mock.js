export const citiesMock = [
  {
    id: 1,
    name: `Paris`,
  },
  {
    id: 2,
    name: `Cologne`,
  },
  {
    id: 3,
    name: `Brussels`,
  },
  {
    id: 4,
    name: `Amsterdam`,
  },
  {
    id: 5,
    name: `Hamburg`,
  },
  {
    id: 6,
    name: `Dusseldorf`,
  },
];

