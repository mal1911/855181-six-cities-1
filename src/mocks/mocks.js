export const offersData = [
  {
    id: 1,
    isPremium: true,
    city: `Paris`,
    title: `Paris Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    isPremium: false,
    city: `Paris`,
    title: `Paris Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    isPremium: false,
    city: `Paris`,
    title: `Paris Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 4,
    isPremium: false,
    city: `Paris`,
    title: `Paris Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 5,
    isPremium: true,
    city: `Cologne`,
    title: `Cologne Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 6,
    isPremium: false,
    city: `Cologne`,
    title: `Cologne Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 7,
    isPremium: true,
    city: `Brussels`,
    title: `Brussels Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 8,
    isPremium: false,
    city: `Brussels`,
    title: `Brussels Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 9,
    isPremium: true,
    city: `Amsterdam`,
    title: `Amsterdam Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 10,
    isPremium: false,
    city: `Amsterdam`,
    title: `Amsterdam Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 11,
    isPremium: true,
    city: `Hamburg`,
    title: `Hamburg Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 12,
    isPremium: false,
    city: `Hamburg`,
    title: `Hamburg Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 13,
    isPremium: true,
    city: `Dusseldorf`,
    title: `Dusseldorf Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 14,
    isPremium: false,
    city: `Dusseldorf`,
    title: `Dusseldorf Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
];

export const citiesData = [
  {
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Cologne`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Brussels`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9],
  },
];
