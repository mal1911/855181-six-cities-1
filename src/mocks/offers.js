export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`
  },
  {
    title: `Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`
  },
  {
    title: `Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`
  },
];
