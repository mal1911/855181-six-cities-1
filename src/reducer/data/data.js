import {transformOfferForLoading, transformCommentsForLoading} from "../../transform-data";
import {HTML_STATUS} from "../../constants";

const initialState = {
  activeCityIndex: 0,
  activeOrderIndex: 0,
  activeOfferId: 0,
  offersData: [],
  favoritesData: [],
  commentsData: [],
  offersError: null,
  favoritesError: null,
  commentsError: null,
  isLoading: true,
};

const ActionType = {
  CHANGE_LOAD_STATUS: `CHANGE_LOAD_STATUS`,
  CHANGE_OFFERS_DATA: `CHANGE_OFFERS_DATA`,
  CHANGE_FAVORITES_DATA: `CHANGE_FAVORITES_DATA`,
  CHANGE_COMMENTS_DATA: `CHANGE_COMMENTS_DATA`,
  CHANGE_ACTIVE_CITY_INDEX: `CHANGE_ACTIVE_CITY_INDEX`,
  CHANGE_ACTIVE_ORDER_INDEX: `CHANGE_ACTIVE_ORDER_INDEX`,
  CHANGE_ACTIVE_OFFER_ID: `CHANGE_ACTIVE_OFFER_ID`,
  CHANGE_OFFERS_ERROR_STATUS: `CHANGE_OFFERS_ERROR_STATUS`,
  CHANGE_FAVORITES_ERROR_STATUS: `CHANGE_FAVORITES_ERROR_STATUS`,
  CHANGE_COMMENTS_ERROR_STATUS: `CHANGE_COMMENTS_ERROR_STATUS`,
};

const ActionCreator = {
  changeLoadStatus: (status) => ({
    type: ActionType.CHANGE_LOAD_STATUS,
    payload: status,
  }),
  changeOffersData: (data) => {
    return {
      type: ActionType.CHANGE_OFFERS_DATA,
      payload: data,
    };
  },
  changeFavoritesData: (data) => {
    return {
      type: ActionType.CHANGE_FAVORITES_DATA,
      payload: data,
    };
  },
  changeCommentsData: (data) => {
    return {
      type: ActionType.CHANGE_COMMENTS_DATA,
      payload: data,
    };
  },
  changeActiveCityIndex: (index) => ({
    type: ActionType.CHANGE_ACTIVE_CITY_INDEX,
    payload: index,
  }),
  changeActiveOrderIndex: (index) => ({
    type: ActionType.CHANGE_ACTIVE_ORDER_INDEX,
    payload: index,
  }),
  changeActiveOfferId: (index) => ({
    type: ActionType.CHANGE_ACTIVE_OFFER_ID,
    payload: index
  }),
  changeOffersErrorStatus: (error) => ({
    type: ActionType.CHANGE_OFFERS_ERROR_STATUS,
    payload: error,
  }),
  changeFavoritesErrorStatus: (error) => ({
    type: ActionType.CHANGE_FAVORITES_ERROR_STATUS,
    payload: error,
  }),
  changeCommentsErrorStatus: (error) => ({
    type: ActionType.CHANGE_COMMENTS_ERROR_STATUS,
    payload: error,
  }),
};


const beforeCommentsLoading = (dispatch) => {
  dispatch(ActionCreator.changeLoadStatus(true));
  dispatch(ActionCreator.changeCommentsErrorStatus(null));
};

const successCommentsLoading = (dispatch, data) => {
  dispatch(ActionCreator.changeCommentsData(
      data.map((obj) => transformCommentsForLoading(obj)))
  );
  dispatch(ActionCreator.changeLoadStatus(false));
};

const errorCommentsLoading = (dispatch, err) => {
  dispatch(ActionCreator.changeCommentsErrorStatus(err));
  dispatch(ActionCreator.changeLoadStatus(false));
};

const beforeOffersLoading = (dispatch) => {
  dispatch(ActionCreator.changeLoadStatus(true));
  dispatch(ActionCreator.changeOffersErrorStatus(null));
  dispatch(ActionCreator.changeFavoritesErrorStatus(null));
};


const Operation = {
  loadOffersData: () => (dispatch, _getState, api) => {
    beforeOffersLoading(dispatch);
    return api.get(`/hotels`)
      .then((response) => {
        const data = response.data.map((obj) => transformOfferForLoading(obj));
        dispatch(ActionCreator.changeOffersData(data));
        dispatch(ActionCreator.changeLoadStatus(false));
      })
      .catch((err) => {
        dispatch(ActionCreator.changeOffersErrorStatus(err));
        dispatch(ActionCreator.changeLoadStatus(false));
      });
  },

  loadFavoritesData: (onGoLogin) => (dispatch, _getState, api) => {
    dispatch(ActionCreator.changeLoadStatus(true));
    dispatch(ActionCreator.changeFavoritesErrorStatus(null));

    return api.get(`/favorite`)
      .then((response) => {
        const data = response.data.map((obj) => transformOfferForLoading(obj));
        dispatch(ActionCreator.changeFavoritesData(data));
        dispatch(ActionCreator.changeLoadStatus(false));
      })
      .catch((err) => {
        if (err.status === HTML_STATUS.FORBIDDEN) {
          onGoLogin();
        } else {
          dispatch(ActionCreator.changeFavoritesErrorStatus(err));
          dispatch(ActionCreator.changeLoadStatus(false));
        }
      });
  },

  changeFavoriteStatus: (id, status, offersData, favoritesData, onGoLogin) => (dispatch, _getState, api) => {
    beforeOffersLoading(dispatch);
    return api.post(`/favorite/${id}/${status}`)
      .then((response) => {
        const newObj = transformOfferForLoading(response.data);
        let newData = offersData.map(
            (obj) => newObj.id === obj.id ? Object.assign({}, newObj) : obj);
        dispatch(ActionCreator.changeOffersData(newData));
        const index = favoritesData.findIndex((obj) => obj.id === id);
        newData = [...favoritesData.slice(0, index), ...favoritesData.slice(index + 1)];
        dispatch(ActionCreator.changeFavoritesData(newData));
        dispatch(ActionCreator.changeLoadStatus(false));
      })
      .catch((err) => {
        if (err.status === HTML_STATUS.FORBIDDEN) {
          onGoLogin();
        } else {
          dispatch(ActionCreator.changeFavoritesErrorStatus(err));
        }
        dispatch(ActionCreator.changeLoadStatus(false));
      });
  },

  loadCommentsData: (id) => (dispatch, _getState, api) => {
    beforeCommentsLoading(dispatch);
    return api.get(`/comments/${id}`)
      .then((response) => {
        successCommentsLoading(dispatch, response.data);
      })
      .catch((err) => {
        errorCommentsLoading(dispatch, err);
      });
  },

  saveCommentObj: (id, commentObj) => (dispatch, _getState, api) => {
    beforeCommentsLoading(dispatch);
    return api.post(`/comments/${id}`, commentObj)
      .then((response) => {
        successCommentsLoading(dispatch, response.data);
      })
      .catch((err) => {
        errorCommentsLoading(dispatch, err);
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_LOAD_STATUS:
      return Object.assign({}, state, {
        isLoading: action.payload
      });
    case ActionType.CHANGE_OFFERS_DATA: {
      return Object.assign({}, state, {
        offersData: action.payload,
      });
    }
    case ActionType.CHANGE_FAVORITES_DATA:
      return Object.assign({}, state, {
        favoritesData: action.payload,
      });
    case ActionType.CHANGE_COMMENTS_DATA:
      return Object.assign({}, state, {
        commentsData: action.payload,
      });
    case ActionType.CHANGE_ACTIVE_CITY_INDEX:
      return Object.assign({}, state, {
        activeCityIndex: action.payload,
      });
    case ActionType.CHANGE_ACTIVE_ORDER_INDEX:
      return Object.assign({}, state, {
        activeOrderIndex: action.payload
      });
    case ActionType.CHANGE_ACTIVE_OFFER_ID: {
      return Object.assign({}, state, {
        activeOfferId: action.payload,
      });
    }
    case ActionType.CHANGE_OFFERS_ERROR_STATUS:
      return Object.assign({}, state, {
        offersError: action.payload
      });
    case ActionType.CHANGE_FAVORITES_ERROR_STATUS:
      return Object.assign({}, state, {
        favoritesError: action.payload
      });
    case ActionType.CHANGE_COMMENTS_ERROR_STATUS:
      return Object.assign({}, state, {
        commentsError: action.payload
      });
  }
  return state;
};

export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
