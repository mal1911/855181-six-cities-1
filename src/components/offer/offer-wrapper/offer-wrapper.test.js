import React from "react";
import renderer from "react-test-renderer";
import thunk from "redux-thunk";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {OfferWrapper} from "./offer-wrapper";
import {offersData, favoritesData, offersLocationsData, initialState} from "../../../mocks/mocks";
import {BrowserRouter} from "react-router-dom";
import {Operation} from '../../../reducer/data/data';

describe(`OfferWrapper correctly renders`, () => {
  const mockStore = configureStore([thunk]);
  let store;
  let tree;
  jest.mock(`../../../reducer/data/data`);
  Operation.loadCommentsData = () => (dispatch) => dispatch(jest.fn());

  beforeEach(() => {
    store = mockStore(initialState);
    tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <OfferWrapper
              offerObj={offersData[0]}
              nearData={offersData.slice(0, 3)}
              offersData={offersData}
              favoritesData={favoritesData}
              nearLocationsData={offersLocationsData.slice(0, 3)}
              error={null}
              onChangeActiveCard={jest.fn()}
              onChangeFavoriteStatus={jest.fn()}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
  });

  it(`OfferWrapper correctly renders`, () => {
    expect(tree).toMatchSnapshot();
  });
});
