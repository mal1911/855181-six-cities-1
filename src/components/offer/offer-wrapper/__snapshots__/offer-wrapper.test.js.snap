// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferWrapper correctly renders OfferWrapper correctly renders 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-02.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-03.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/room.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-02.jpg"
          />
        </div>
      </div>
    </div>
    ;
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Beautiful & luxurious apartment at great location
          </h1>
          <button
            className="property__bookmark-button button property__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Entire 
            room
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            1
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            2
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            231
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Towels
            </li>
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
            >
              Air conditioning
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
            <li
              className="property__inside-item"
            >
              Baby seat
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="/img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              0
            </span>
          </h2>
          <ul
            className="reviews__list"
          />
        </section>
      </div>
      <section
        className="property__map map"
        id="map"
      />
    </div>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="near__place-card place-card"
        >
          <div
            className="near__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  264
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button    button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                aria-current={null}
                href="/offer/4"
                onClick={[Function]}
              >
                Penthouse, 4-5 rooms + 5 balconies
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              room
            </p>
          </div>
        </article>
        <article
          className="near__place-card place-card"
        >
          <div
            className="near__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  253
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button    button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                aria-current={null}
                href="/offer/10"
                onClick={[Function]}
              >
                Penthouse, 4-5 rooms + 5 balconies
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              room
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
